Some suggestions from Clang-tidy

thalieht opened the pull request:
- Don't use 'else' after return/break
- Pass by value and use std::move in some constructors
- Convert to range-based for loop
- Use 'auto' to avoid duplicating the type name
- Use isEmpty() instead of size() for emptiness 
- Convert vars into refs to avoid copy construction from const refs
- Replace deprecated header 'signal.h' with 'csignal' 
- Avoid repeating the return type
- Remove some unused stuff

glassez commented on src/base/bittorrent/peerinfo.cpp:
```diff
@@ -51,8 +51,8 @@ PeerAddress::PeerAddress(const QHostAddress &ip, ushort port)
-PeerInfo::PeerInfo(const TorrentHandle *torrent, const libt::peer_info &nativeInfo)
-    : m_nativeInfo(nativeInfo)
+PeerInfo::PeerInfo(const TorrentHandle *torrent, libt::peer_info nativeInfo)
+    : m_nativeInfo(std::move(nativeInfo))
```
Sorry, I don't understand this change. And all the similar. What's it for?

thalieht replied:
I read somewhere that it costs 2 moves instead of 1 copy and 1 move but turns out it's for rvalues... i'll just drop this commit.

glassez replied:
>i'll just drop this commit.

👍

Chocobo1 commented on src/webui/webapplication.cpp:
```diff
@@ -512,8 +512,8 @@ QString WebApplication::generateSid() const
+        for (auto &entry : tmp)
+            entry = Utils::Random::rand();
 
         sid = QByteArray::fromRawData(reinterpret_cast<const char *>(tmp), sizeof(quint32) * size).toBase64();
```
Or rewrite this block to:
```c++
const quint32 tmp[] = {Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand()};
sid = QByteArray::fromRawData(reinterpret_cast<const char *>(tmp), sizeof(tmp)).toBase64();
```

thalieht replied:
Would `using namespace Utils;` be acceptable here to reduce the length?

Chocobo1 replied:
>Would `using namespace Utils;` be acceptable here to reduce the length?

should be ok, however I don't really think there is such need, seldom we need to read that line carefully.
here is a even shorter version:
```c++
const auto r = []() { return Utils::Random::rand(); };
const quint32 tmp[] = {r(), r(), r(), r(), r(), r()};
sid = QByteArray::fromRawData(reinterpret_cast<const char *>(tmp), sizeof(tmp)).toBase64();
```

glassez replied:
>here is a even shorter version:

Do you really think it's a good idea to wrap it in another callable? Isn't that too wasteful just to shorten its name?
I'd use `std::generate`, but I'm afraid it'll break your "peace of mind", because you'll have to have "tmp" mutable.

Chocobo1 replied:
>Isn't that too wasteful just to shorten its name?

Wasteful in what? one more alias or does it incur performance penalty?
As it is not much difference from using algorithm functions in std lib, I would expect the compiler to inline the inner of lambda and have zero penalty.

thalieht replied:
I picked the original without "using...". I was gonna split the line anyway.

glassez replied:
Both old code and `std::generate` are quite good scalable algorithms. Sorry, but IMO your suggested code is a special case of absurdity aimed only at making "tmp" variable to be constant.
I agree that having many variables immutable is a good tool to avoid some possible errors. But it seems that for you it is not a "tool", but a "goal in itself". And you are ready to achieve it regardless of the costs. It really bothers me.

glassez replied:
@Chocobo1, I apologize in advance if my comment above seems offensive to you. It's nothing personal. But once you offer to go by questionable way here, please bother to prove its advantages.

Chocobo1 replied:
>is a special case of absurdity aimed only at making "tmp" variable to be constant.

It is a special case... because only 6 function calls are required and since it is short why not make the tmp const too? If 100 calls  are needed then std::generate will be suitable IMO.
I think this is merely personal preference and you over reacted.

@thalieht 
I don't insist on my suggestion, pick one as you see fit.

glassez replied:
>I think this is merely personal preference

Following personal preference can be accepted when you produce new code. But making changes of existing things based on subjective preferences is counterproductive. Such an approach can generate a lot of useless changes that simply adjust some part of the code to someone's personal preferences, without bringing any objective benefit. Am I right?

Chocobo1 commented on src/gui/transferlistmodel.cpp:
```diff
@@ -406,7 +406,7 @@ QColor getColorByState(BitTorrent::TorrentState state)
     default:
         Q_ASSERT(false);
-        return QColor(255, 0, 0); // red
+        return {255, 0, 0}; // red
```
you might also apply to this whole function, I mean to other `return QColor(...);`

thalieht replied:
I made a search in the project and found a whole lot more of these but i don't know if there's a reason clang-tidy didn't mention them... added new commit with them.

Chocobo1 commented:
Some personal wishes you might be interested to include in this PR:
1. remove an extra empty line in torrentfileguard.cpp, before the end of file
2. typo in torrentfileguard.h: `TOFFENTFILEGURAD_H` -> `TORRENTFILEGUARD_H`, or use `#pragma once`.

Chocobo1 commented on src/gui/mainwindow.cpp:
```diff
@@ -795,7 +795,7 @@ void MainWindow::cleanup()
 
     // remove all child widgets
-    while (QWidget *w = findChild<QWidget * >())
+    while (auto *w = findChild<QWidget * >())
```
`* >` -> `*>`

thalieht commented:
Removed commit about "passing by value + std::move" and added the following to "Remove some unused stuff" commit.
>* remove an extra empty line in torrentfileguard.cpp, before the end of file
>* typo in torrentfileguard.h: `TOFFENTFILEGURAD_H` -> `TORRENTFILEGUARD_H`, or use `#pragma once`.

glassez commented on src/gui/fspathedit_p.cpp:
```diff
@@ -251,8 +251,7 @@ void Private::FileLineEdit::keyPressEvent(QKeyEvent *e)
 
-    const FileSystemPathValidator *validator =
-        qobject_cast<const FileSystemPathValidator *>(this->validator());
+    const auto *validator = qobject_cast<const FileSystemPathValidator *>(this->validator());
```
Sorry, I lost your previous comment, so I can't quote it... But I guess you didn't know what I meant.
Since you have `qobject_cast<const FileSystemPathValidator *>` at right hand both `const auto *validator` and `auto *validator` (and even `auto validator`) are deduced to `const FileSystemPathValidator *` so `const` at left is redundant!

thalieht replied:
And i was wondering where this comment went. Thanks for the explanation, i'll remove it.

Kolcha commented on src/gui/properties/trackerlistwidget.cpp:
```diff
@@ -466,8 +466,7 @@ void TrackerListWidget::editSelectedTracker()
     bool match = false;
-    for (int i = 0; i < trackers.size(); ++i) {
-        BitTorrent::TrackerEntry &entry = trackers[i];
+    for (auto &entry : trackers) {
```
beware when using similar loops, container can be detached!
from [Qt documentation](https://doc.qt.io/qt-5/containers.html#the-foreach-keyword):
> An alternative to Qt's foreach loop is the range-based for that is part of C++ 11 and newer. However, keep in mind that the range-based for might force a Qt container to detach, whereas foreach would not. But using foreach always copies the container, which is usually not cheap for STL containers. If in doubt, prefer foreach for Qt containers, and range based for for STL ones.

but from [`foreach` documentation](https://doc.qt.io/qt-5/qtglobal.html#foreach):
> Note: Since Qt 5.7, the use of this macro is discouraged. It will be removed in a future version of Qt. Please use C++11 range-for, possibly with qAsConst(), as needed.

so, as result [`qAsConst()`](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) must be used in similar cases to prevent container detaching. see [its documentation](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) for examples, they have a lot of useful comments

Kolcha commented on src/gui/properties/trackerlistwidget.cpp:
```diff
@@ -466,8 +466,7 @@ void TrackerListWidget::editSelectedTracker()
     bool match = false;
-    for (int i = 0; i < trackers.size(); ++i) {
-        BitTorrent::TrackerEntry &entry = trackers[i];
+    for (auto &entry : trackers) {
```
beware when using similar loops, container can be detached!
from [Qt documentation](https://doc.qt.io/qt-5/containers.html#the-foreach-keyword):
> An alternative to Qt's foreach loop is the range-based for that is part of C++ 11 and newer. However, keep in mind that the range-based for might force a Qt container to detach, whereas foreach would not. But using foreach always copies the container, which is usually not cheap for STL containers. If in doubt, prefer foreach for Qt containers, and range based for for STL ones.

but from [`foreach` documentation](https://doc.qt.io/qt-5/qtglobal.html#foreach):
> Note: Since Qt 5.7, the use of this macro is discouraged. It will be removed in a future version of Qt. Please use C++11 range-for, possibly with qAsConst(), as needed.

so, as result [`qAsConst()`](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) must be used in similar cases to prevent container detaching. see [its documentation](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) for examples, they have a lot of useful comments

glassez replied:
@Kolcha, how your comment is applicaple to this particular place?

Kolcha replied:
sorry, I didn't see line `entry = newEntry;` below... so, there no any issues in this case. I'll delete my my comment later

Chocobo1 commented:
Thank you!

Chocobo1 commented on src/gui/transferlistmodel.cpp:
```diff
@@ -149,22 +149,22 @@ QVariant TransferListModel::headerData(int section, Qt::Orientation orientation,
             case TR_PRIORITY:
             case TR_LAST_ACTIVITY:
-                return QVariant(Qt::AlignRight | Qt::AlignVCenter);
+                return {Qt::AlignRight | Qt::AlignVCenter};
```
@thalieht 
This line introduced a warning: 
https://ci.appveyor.com/project/qbittorrent/qbittorrent/builds/22423194#L541

Could you revert it?

thalieht replied:
Sure.

thalieht replied:
Reverted in #10314
